flow/model/qvalue/avro_converter.go

```diff
@@ -205,103 +212,126 @@ func QValueToAvro(
 			(len(v.Value().(string)) > 15*1024*1024) {
 			slog.WarnContext(ctx, "Clearing TEXT value > 15MB for Snowflake!")
 			slog.WarnContext(ctx, "Check this issue for details: https://github.com/PeerDB-io/peerdb/issues/309")
+
 			if c.Nullable {
-				return nil, nil
+				return nil, size, nil
 			} else {
-				return "", nil
+				return "", size, nil
 			}
 		}
-		return c.processNullableUnion(v.Value()), nil
+		return c.processNullableUnion(v.Value()), size, nil
 	case types.QValueFloat32:
+		size := constSize(4, nullableSize, calcSize)
 		if c.TargetDWH == protos.DBType_BIGQUERY {
-			return c.processNullableUnion(float64(v.Val)), nil
+			return c.processNullableUnion(float64(v.Val)), size, nil
 		}
-		return c.processNullableUnion(v.Val), nil
+		return c.processNullableUnion(v.Val), size, nil
 	case types.QValueFloat64:
-		return c.processNullableUnion(v.Val), nil
+		return c.processNullableUnion(v.Val), constSize(8, nullableSize, calcSize), nil
 	case types.QValueInt8:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueInt16:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueInt32:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueInt64:
-		return c.processNullableUnion(v.Val), nil
+		return c.processNullableUnion(v.Val), varIntSize(v.Val, nullableSize, calcSize), nil
 	case types.QValueInt256:
-		return c.processInt256(v.Val), nil
+		return c.processInt256(v.Val), constSize(32, nullableSize, calcSize), nil
 	case types.QValueUInt8:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueUInt16:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueUInt32:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueUInt64:
-		return c.processNullableUnion(int64(v.Val)), nil
+		return c.processNullableUnion(int64(v.Val)), varIntSize(int64(v.Val), nullableSize, calcSize), nil
 	case types.QValueUInt256:
-		return c.processUInt256(v.Val), nil
+		return c.processUInt256(v.Val), constSize(32, nullableSize, calcSize), nil
 	case types.QValueBoolean:
-		return c.processNullableUnion(v.Val), nil
+		return c.processNullableUnion(v.Val), constSize(1, nullableSize, calcSize), nil
 	case types.QValueNumeric:
-		return c.processNumeric(v.Val), nil
+		return c.processNumeric(v.Val), stringSize(v.Val.String(), nullableSize, calcSize), nil
```

Comment:
If the implementation is not planned, please make a task to push the calculation to processNumeric() and a TODO for when flattened mode comes with Decimal128 or if we decide to use it for PG. Enabling this for PG today would mean double perf hit on long numerics (which is the heaviest type to process).